errornotifier/airbraker.go

```diff
@@ -6,22 +6,31 @@ package errornotifier
 import (
 	"errors"
 	"fmt"
+	"io"
 	"net/http"
 	"runtime/debug"
 
+	"github.com/airbrake/gobrake"
 	"github.com/theplant/appkit/log"
-	gobrake "gopkg.in/airbrake/gobrake.v1"
 )
 
 // Notifier defines an interface for reporting error
 type Notifier interface {
-	Notify(interface{}, *http.Request) error
+	Notify(err interface{}, r *http.Request, context map[string]interface{})
```

Comment:
One note about this way:

If we use the Notifier to notify an error inside the request handler manually, then we need to put the `RequestID` or `TraceID` in manually, then the report will contains that information. 

It's nice to have that sending airbrake inside an handler will have the `RequestID` and `TraceID` included automatically.

Reply:
After discussed with @lifenod for a while, in Backend, the usage of Airbrake is the http server, when http request failed, the airbrake will send out. And no other usage of Airbrake in the http handlers, so I think we even with:

> If we use the Notifier to notify an error inside the request handler manually, then we need to put the RequestID or TraceID in manually, then the report will contains that information.

The result is the same. 

Let's forget this.

Reply:
I think this is good because you can add extra data if you need to.